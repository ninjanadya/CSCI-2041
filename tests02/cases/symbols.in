(define n
  (+ 4 5))

(define m
  (if (= n 4)
    1
    (* n 4)))

(define -
  (lambda (m n)
    (- m n)
  )
)

