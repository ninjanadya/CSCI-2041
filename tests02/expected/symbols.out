OpenParenToken
SymbolToken define
SymbolToken n
OpenParenToken
SymbolToken +
NumberToken 4
NumberToken 5
CloseParenToken
CloseParenToken
OpenParenToken
SymbolToken define
SymbolToken m
OpenParenToken
SymbolToken if
OpenParenToken
SymbolToken =
SymbolToken n
NumberToken 4
CloseParenToken
NumberToken 1
OpenParenToken
SymbolToken *
SymbolToken n
NumberToken 4
CloseParenToken
CloseParenToken
CloseParenToken
OpenParenToken
SymbolToken define
SymbolToken -
OpenParenToken
SymbolToken lambda
OpenParenToken
SymbolToken m
SymbolToken n
CloseParenToken
OpenParenToken
SymbolToken -
SymbolToken m
SymbolToken n
CloseParenToken
CloseParenToken
CloseParenToken
EndToken
